lib/rucio/core/config.py

```diff
@@ -423,7 +429,32 @@ def remove_option(section: str, option: str, *, session: "Session") -> bool:
         return True
 
 
-def read_from_cache(key: str, expiration_time: int = 900) -> Any:
+@overload
+def read_from_cache(key: Literal['sections'], expiration_time: int = 900) -> Union[list[str], NoValue]:
+    ...
+
+
+@overload
+def read_from_cache(key: OptionsKey, expiration_time: int = 900) -> Union[list[str], NoValue]:
+    ...
```

Comment:
These two have the same argument list and output type, so they could be unified into a single overload:

```suggestion
def read_from_cache(key: Union[OptionsKey, Literal['sections']], expiration_time: int = 900) -> Union[list[str], NoValue]:
    ...
```